### ISSUE TITLE
Add support for :default option

### ISSUE THREAD

**Issue by skydan** [palkan/store_attribute#6]
Does it make sense to add support for `:default` option on top of set of typecasting options?

**Comment by palkan** [comment]
I think, it makes sense.

Though it's not clear how exactly we want it to work. For example:

```ruby
class Product < ActiveRecord::Base
   store_attribute :settings, :color, :string, default: "red"
end
```

How this should work?
- accessor returns the default value (seems reasonable)

```ruby
Product.new.color #=> "red"
```

-  should store attribute contain the default?

```ruby
Product.new.settings #=> ? should this be `{ "color" => "red"}` or `{}`?
```

- should we persist the default value on save?

```ruby
Product.new.save!
#=> INSERT INTO products (settings) values ('{"color":"red"}'); ???
```

**Comment by skydan** [comment]
> How this should work?
> 
>   * accessor returns the default value (seems reasonable)
> 
> 
> ```ruby
> Product.new.color #=> "red"
> ```
> 
>   * should store attribute contain the default?
> 
> 
> ```ruby
> Product.new.settings #=> ? should this be `{ "color" => "red"}` or `{}`?
> ```
> 
>   * should we persist the default value on save?
> 
> 
> ```ruby
> Product.new.save!
> #=> INSERT INTO products (settings) values ('{"color":"red"}'); ???
> ```

I think that behavior should be the same as we pass this attribute explicitly:
```ruby
Product.new.settings #=>`{ "color" => "red"}`
Product.new(color: "red").settings #=>`{ "color" => "red"}`

Product.new.save!
#=> INSERT INTO products (settings) values ('{"color":"red"}'); 
Product.new(color: "red").save!
#=> INSERT INTO products (settings) values ('{"color":"red"}'); 
```

**Comment by palkan** [comment]
I think, we need a parity with Attributes API here:

```ruby
# create_table :posts do |t|
#   t.string :data
# end
class Post < ActiveRecord::Base
  attribute :data, :datetime, default: -> { Time.current }
end
```

How does this behave? Not sure, need to check.

**Comment by theycallmeswift** [comment]
+1 that this would be valuable and should mirror the attributes API.

**Comment by SumLare** [comment]
I'm gonna take a look at it this week if nobody minds.

**Comment by palkan** [comment]
Closed by #7 and released in 0.6.0
